app/controllers/discourse_login_client/auth_controller.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module DiscourseLoginClient
+  class AuthController < ApplicationController
+    requires_plugin "discourse-login-client"
+
+    skip_before_action :verify_authenticity_token, only: [:revoke]
+
+    def revoke
+      signature = params.require(:signature)
+      user_id = params.require(:user_id)
+      timestamp = params.require(:timestamp)
+
+      RateLimiter.new(nil, "discourse_login_revoke_#{user_id}", 5, 1.minute).performed!
+
+      time_diff = (Time.now.to_i - timestamp.to_i).abs
+      if time_diff > 5.minutes.to_i
+        if SiteSetting.discourse_login_debug_auth
```

Comment:
For a follow-up, my mistake here, this setting should be called `discourse_login_client_debug_auth`.

Reply:
Yeah, it's not the best name. Will change in another PR.